addons/website/static/src/snippets/s_instagram_page/000.js

```diff
@@ -0,0 +1,135 @@
+        iframeEl.setAttribute("scrolling", "no");
+        // We can already estimate the height of the iframe.
+        iframeEl.height = this._estimateIframeHeight();
+        // We have to setup the message listener before setting the src, because
+        // the iframe can send a message before this JS is fully loaded.
+        this.__onMessage = this._onMessage.bind(this);
+        window.addEventListener("message", this.__onMessage);
+        // We set the src now, we are ready to receive the message.
+        iframeEl.src = `https://www.instagram.com/${this.$target[0].dataset.instagramPage}/embed`;
+
+        return this._super(...arguments);
+    },
+    /**
+     * @override
+     */
+    destroy() {
+        window.removeEventListener("resize", this.__onResize);
+        const iframeEl = this.$target[0].querySelector(".o_instagram_container iframe");
+        if (iframeEl) {
+            iframeEl.remove();
+            window.removeEventListener("message", this.__onMessage);
+        }
+        this._super.apply(this, arguments);
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Gives an estimation of the height of the Instagram iframe.
+     *
+     * @private
+     * @returns {number}
+     */
+    _estimateIframeHeight() {
+        // In the meantime Instagram doesn't send us a message with the height,
+        // we use a formula to estimate the height of the iframe (the formula
+        // has been found with a linear regression).
+        const iframeEl = this.$target[0].querySelector(".o_instagram_container iframe");
+        const iframeWidth = parseInt(getComputedStyle(iframeEl).width);
+        if (iframeWidth < 432) {
+            // The profile picture is smaller when width < 432px.
+            return 0.659 * iframeWidth + 156;
+        }
+        return 0.659 * iframeWidth + 203;
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when a message is sent. Instagram sends us a message with the
+     * height of the iframe.
+     *
+     * @private
+     * @param {Event} ev
+     */
+    _onMessage(ev) {
+        const iframeEl = this.$target[0].querySelector(".o_instagram_container iframe");
+        if (!iframeEl) {
+            // TODO: fix this case. We should never end up here. It happens when
+            // - Drop Instagram snippet
+            // - Undo
+            // - Drop a new Instagram snippet
+            // => The listener of the first one is still active because the
+            // public widget has not been destroyed.
+            // GREP: PUBLIC_WIDGET_NOT_DESTROYED
+            window.removeEventListener("message", this.__onMessage);
+            return;
+        }
```

Comment:
I created a task to fix this issue of public widget not destroyed on undo.
[Task-3508448](https://www.odoo.com/web#id=3508448&cids=1&menu_id=4720&action=333&active_id=974&model=project.task&view_type=form)